src/main/java/com/rpg2014/invokers/EC2Invoker.java

```diff
@@ -1,24 +1,54 @@
 package com.rpg2014.invokers;
 
+import com.rpg2014.model.Ec2MethodNames;
 import com.rpg2014.wrappers.SpidermanEC2Wrapper;
+
+import javax.ws.rs.InternalServerErrorException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 public class EC2Invoker {
 
-    private static SpidermanEC2Wrapper instance = SpidermanEC2Wrapper.getInstance();
+    Executor executor = Executors.newSingleThreadExecutor();
+    private SpidermanEC2Wrapper instance = SpidermanEC2Wrapper.getInstance();
+
+    public Optional invoke(Ec2MethodNames methodName) {
+        Runnable r;
+        switch (methodName) {
+            case StartInstance:
+            case StopInstance:
+                r = getAsyncRunnable(methodName);
+                executor.execute(r);
+                return Optional.empty();
+            default:
+                return invokeMethod(methodName);
+        }
+    }
 
-    public static void invoker(String methodName) throws InterruptedException {
-        Runnable r = () -> {
+    private Runnable getAsyncRunnable(Ec2MethodNames methodName){
+        return () -> {
             Method method = null;
             try {
-                method = instance.getClass().getDeclaredMethod(methodName);
+                method = instance.getClass().getDeclaredMethod(methodName.toString());
```

Comment:
methodName is passed in as a string already

Reply:
Nah, its an enum now.  but this needs to change to .getMethodName() anyways